apps/mobile/src/screens/onboarding/onboarding.tsx

```diff
@@ -0,0 +1,213 @@
+import { type NativeStackNavigationProp } from "@react-navigation/native-stack";
+import InfinitePager, {
+	type InfinitePagerImperativeApi,
+} from "react-native-infinite-pager";
+import {
+	type OnboardingAnswerRequestBodyDto,
+	oneAnswerSelectedValidationSchema,
+} from "shared";
+
+import {
+	BackgroundWrapper,
+	Button,
+	LoaderWrapper,
+	ProgressBar,
+	ScreenWrapper,
+	View,
+} from "~/libs/components/components";
+import { PREVIOUS_INDEX_OFFSET } from "~/libs/constants/constants";
+import { DataStatus, RootScreenName } from "~/libs/enums/enums";
+import {
+	useAppDispatch,
+	useAppForm,
+	useAppSelector,
+	useCallback,
+	useEffect,
+	useMemo,
+	useNavigation,
+	useRef,
+	useState,
+} from "~/libs/hooks/hooks";
+import { actions as onboardingActions } from "~/slices/onboarding/onboarding";
+
+import { pageInterpolatorSlide } from "./libs/animations/animations";
+import { Content } from "./libs/components/components";
+import {
+	ONBOARDING_FORM_DEFAULT_VALUES,
+	ONE,
+	ZERO,
+} from "./libs/constants/constants";
+import { type OnboardingFormValues } from "./libs/types/types";
+import { styles } from "./styles";
+
+type RootStackParameterList = {
+	[RootScreenName.WELCOME]: undefined;
+};
+
+const Onboarding: React.FC = () => {
+	const dispatch = useAppDispatch();
+	const pagerReference = useRef<InfinitePagerImperativeApi | null>(null);
+
+	const navigation =
+		useNavigation<NativeStackNavigationProp<RootStackParameterList>>();
+
+	const [answerIds, setAnswerIds] = useState<OnboardingAnswerRequestBodyDto>({
```

Comment:
Just curious, is it possible to create a selector like `selectAnswersByQuestionIndex` and get rid of this useState and complex switching logic?